sdk/go/common/resource/plugin/langruntime_plugin.go

```diff
@@ -521,19 +531,29 @@ func RPCDiagnosticToHclDiagnostic(diag *codegenrpc.Diagnostic) *hcl.Diagnostic {
 		}
 	}
 
-	return &hcl.Diagnostic{
-		Severity: hcl.DiagnosticSeverity(diag.Severity),
-		Summary:  diag.Summary,
-		Detail:   diag.Detail,
-		Subject: &hcl.Range{
+	var subject *hcl.Range
+	if diag.Subject != nil {
+		subject = &hcl.Range{
 			Filename: diag.Subject.Filename,
 			Start:    rpcPosToPos(diag.Subject.Start),
 			End:      rpcPosToPos(diag.Subject.End),
-		},
-		Context: &hcl.Range{
+		}
+	}
+
+	var context *hcl.Range
+	if diag.Subject != nil {
```

Comment:
Same as above

Reply:
Thankfully the tests also picked up on this 😆